sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastHashJoinExecBase.scala

```diff
@@ -115,6 +139,14 @@ abstract class GpuBroadcastHashJoinExecBase(
     STREAM_TIME -> createNanoTimingMetric(DEBUG_LEVEL, DESCRIPTION_STREAM_TIME),
     JOIN_TIME -> createNanoTimingMetric(DEBUG_LEVEL, DESCRIPTION_JOIN_TIME))
 
+  override lazy val (buildOutput, streamedOutput, joinLeftOutput, joinRightOutput) =
+    nonAstJoinCondSplit match {
+    case asProj: JoinCondSplitAsProject =>
```

Comment:
This is an indication that the abstraction layers have broken down and are insufficient.  Seems like the strategy abstraction class should have interfaces to handle this without examining class types.

Reply:
Refactor by making split strategy as an attribute to ```GpuHashJoin```.